state/resources_persistence_staged.go

```diff
@@ -92,7 +92,7 @@ func (staged StagedResource) Activate() error {
 				logger.Errorf("can't read existing resource during activate: %v", errors.Details(err))
 				return nil, errors.Trace(err)
 			}
-			if hasNewBytes {
+			if hasNewBytes && incrementCharmModifiedVersion == IncrementCharmModifiedVersion {
```

Comment:
since this is a bool, we can just do
`if hasNewBytes && incrementCharmModifiedVersion {`

Reply:
incrementCharmModifiedVersion is bool but the type of it is `IncrementCharmModifiedVersionType`.
so if hasNewBytes && incrementCharmModifiedVersion will get type mismatch error.